compiler/src/symboltable.rs

```diff
@@ -560,9 +560,17 @@ impl SymbolTableBuilder {
             }
             Bytes { .. } => {}
             Tuple { elements } | Set { elements } | List { elements } | Slice { elements } => {
-                self.scan_expressions(elements, &ExpressionContext::Load)?;
+                self.scan_expressions(elements, context)?;
             }
             Comprehension { kind, generators } => {
+                for generator in generators {
```

Comment:
Was there a bug that this block is moved? Btw, generators are implemented as functions, so it might make sense to define a new symbol table for generators. We could check with the crawl_source.py script on the test snippet you added here.

Reply:
Reverted it because it didn't afffect the issue